File: labs_python/lab4/SmartHomeUI.py
from SmartHomeController import SmartHomeController

def SmartHomeUI():
    controller = SmartHomeController()
    house = "MyHouse"  # Placeholder for the house name
    exit_system = False

    print("Welcome to the Smart Home System!")
    while not exit_system:
        print("\n=== Smart Home Menu ===")
        print("1. Control Lights")
        print("2. Control Heating")
        print("3. Control Security")
        print("4. Check System Status")
        print("5. Exit")
        choice = input("Select an option: ")

        if choice == "1":
            light_choice = input("1. Turn ON Lights\n2. Turn OFF Lights\nChoose an option: ")
            if light_choice == "1":
                controller.control_lights(True)
            elif light_choice == "2":
                controller.control_lights(False)
            else:
                print("Invalid option.")

        elif choice == "2":
            heating_choice = input("1. Turn ON Heating\n2. Turn OFF Heating\nChoose an option: ")
            if heating_choice == "1":
                controller.control_heating(True)
            elif heating_choice == "2":
                controller.control_heating(False)
            else:
                print("Invalid option.")

        elif choice == "3":
            security_choice = input("1. Activate Security\n2. Deactivate Security\nChoose an option: ")
            if security_choice == "1":
                controller.control_security(True)
            elif security_choice == "2":
                controller.control_security(False)
            else:
                print("Invalid option.")

        elif choice == "4":
            print(controller.check_system_status())

        elif choice == "5":
            exit_system = True
            print("Exiting Smart Home System. Goodbye!")

        else:
            print("Invalid choice. Please try again.")

File: labs_java/lab4/SmartHomeUI.java
import java.util.Scanner;

public class SmartHomeUI {
    public static void main(String[] args) {
        SmartHomeController controller = SmartHomeController.getInstance();
        Scanner scanner = new Scanner(System.in);

        String house = "MyHouse"; // Назва будинку
        boolean exit = false;

        System.out.println("Welcome to Smart Home System!");
        while (!exit) {
            System.out.println("\n +++ Smart Home Menu +++");
            System.out.println("1. Control Lights");
            System.out.println("2. Control Heating");
            System.out.println("3. Control Security");
            System.out.println("4. Check System Status");
            System.out.println("5. Exit");
            System.out.print("Select an option: ");

            int choice = scanner.nextInt();

            switch (choice) {
                case 1:
                    System.out.println("1. Turn ON Lights");
                    System.out.println("2. Turn OFF Lights");
                    System.out.print("Choose an option: ");
                    int lightChoice = scanner.nextInt();
                    controller.controlLights(house, lightChoice == 1);
                    break;

                case 2:
                    System.out.println("1. Turn ON Heating");
                    System.out.println("2. Turn OFF Heating");
                    System.out.print("Choose an option: ");
                    int heatingChoice = scanner.nextInt();
                    controller.controlHeating(house, heatingChoice == 1);
                    break;

                case 3:
                    System.out.println("1. Activate Security");
                    System.out.println("2. Deactivate Security");
                    System.out.print("Choose an option: ");
                    int securityChoice = scanner.nextInt();
                    controller.controlSecurity(house, securityChoice == 1);
                    break;

                case 4:
                    System.out.println(controller.checkSystemStatus(house));
                    break;

                case 5:
                    exit = true;
                    System.out.println("Exiting Smart Home System. Goodbye!");
                    break;

                default:
                    System.out.println("Invalid choice. Please try again.");
            }
        }

        scanner.close();
    }
}
